src/OMSimulatorLib/Model.cpp

```diff
@@ -678,6 +679,21 @@ oms_status_enu_t oms::Model::addResources(const oms::ComRef& cref, const std::st
     fileName = path_.filename().generic_string();
   }
 
+  // TODO: this is a temporary workaround, we need a proper helper function to get the extension, .srmd is bigger that 4 so the old check is not valid
+  std::string extension = filesystem::path(fileName).extension().generic_string();
+
+  // initialize validator
+  oms::XercesValidator validator;
+
+  // validate the srmd file
+  if (extension == ".srmd")
+  {
+    oms_status_enu_t status = validator.validateSRMD(path_);
```

Comment:
`path_` needs to be converted to  ` path_.generic_string()`, or change the function argument in `validateSRMD` to fileSystem::Path